proxy-robots-module/src/main/java/orwell/proxy/Utils.java

```diff
@@ -9,4 +13,34 @@
         return result;
     }
 
+    /**
+     * Split a byte array around a byte separator, within a given limit
+     *
+     * @param separator byte separator
+     * @param input     bytes array to split
+     * @param limit     max number of elements allowed in the returned list
+     *                  If limit is < 0, then proceed as if there were no limit
+     * @return list of bytes arrays, it size being of maximum {limit} elements
+     */
+    public static List<byte[]> split(final byte separator, final byte[] input, final int limit) {
+        final List<byte[]> list = new LinkedList<>();
+        if (0 == limit) {
+            return list;
+        }
+
+        int blockStart = 0;
+        if (1 != limit) {
```

Comment:
Actually, no.

if limit = 0: returns a new empty list
if limit = 1: returns the first element result of the split (more precisely: a list of one byte array of all the bytes encountered in the input until the first separator)

I am adding this comment in the javadoc